airflow/models/taskinstance.py

```diff
@@ -3000,8 +3019,8 @@ def _execute_task(self, context: Context, task_orig: Operator):
         return _execute_task(self, context, task_orig)
 
     @provide_session
-    def defer_task(self, exception: TaskDeferred, session: Session) -> None:
-        """Mark the task as deferred and sets up the trigger that is needed to resume it.
+    def defer_task(self, exception: TaskDeferred | None = None, session: Session = NEW_SESSION) -> None:
```

Comment:
```suggestion
    def defer_task(self, exception: TaskDeferred | None, session: Session = NEW_SESSION) -> None:
```

Let’s make this a required argument for explicitness.

Reply:
sounds good! just updated it